djvu.c

```diff
@@ -162,6 +162,32 @@ static int setColorRendering(lua_State *L) {
 	return 0;
 }
 
+static int getMetadata(lua_State *L) {
+	DjvuDocument *doc = (DjvuDocument*) luaL_checkudata(L, 1, "djvudocument");
+	miniexp_t anno = ddjvu_document_get_anno(doc->doc_ref, 1);
+	miniexp_t *keys = ddjvu_anno_get_metadata_keys(anno);
+
+	// `keys` can be null if there's an error. In that case,
+	// we have the choice of returning either `nil` or
+	// the empty table back to lua.  Here we prefer the latter.
+	lua_newtable(L);
+	if (!keys) return 1;
```

Comment:
This was bothering me. The docs indicate that `keys` should never be null; however, looking at the source `malloc` can fail to allocate our array---which is quite obvious in retrospect.

To that effect, I ammended the original PR with this extra null check. Since we have a choice on how to handle it and the ordering with `lua_newtable` is important, I went ahead and threw in a comment.